src/Sylius/Behat/Context/Api/Admin/ManagingChannelPriceHistoryConfigContext.php

```diff
@@ -0,0 +1,195 @@
+    ) {
+    }
+
+    /**
+     * @When /^I (enable|disable) showing the lowest price of discounted products$/
+     */
+    public function iEnableShowingTheLowestPriceOfDiscountedProducts(string $visible): void
+    {
+        $this->client->addRequestData(
+            'channelPriceHistoryConfig',
+            ['lowestPriceForDiscountedProductsVisible' => $visible === 'enable'],
+        );
+    }
+
+    /**
+     * @When /^I specify (-?\d+) days as the lowest price for discounted products checking period$/
+     */
+    public function iSpecifyDaysAsTheLowestPriceForDiscountedProductsCheckingPeriod(int $days): void
+    {
+        $this->client->addRequestData(
+            'channelPriceHistoryConfig',
+            ['lowestPriceForDiscountedProductsCheckingPeriod' => $days],
+        );
+    }
+
+    /**
+     * @When I exclude the :taxon taxon from showing the lowest price of discounted products
+     */
+    public function iExcludeTheTaxonFromShowingTheLowestPriceOfDiscountedProducts(TaxonInterface $taxon): void
+    {
+        $this->iExcludeTheTaxonsFromShowingTheLowestPriceOfDiscountedProducts([$taxon]);
+    }
+
+    /**
+     * @When I remove the :taxon taxon from excluded taxons from showing the lowest price of discounted products
+     */
+    public function iRemoveTheTaxonFromExcludedTaxonsFromShowingTheLowestPriceOfDiscountedProducts(TaxonInterface $taxon): void
+    {
+        /** @var ChannelInterface $channel */
+        $channel = $this->sharedStorage->get('channel');
+
+        $leftTaxons = [];
+        foreach ($channel->getChannelPriceHistoryConfig()->getTaxonsExcludedFromShowingLowestPrice() as $excludedTaxon) {
+            if ($excludedTaxon->getId() !== $taxon->getId()) {
+                $leftTaxons[] = $excludedTaxon;
+            }
+        }
+
+        $this->iExcludeTheTaxonsFromShowingTheLowestPriceOfDiscountedProducts($leftTaxons);
+    }
+
+    /**
+     * @When /^I exclude the ("[^"]+" and "[^"]+" taxons) from showing the lowest price of discounted products$/
+     */
+    public function iExcludeTheTaxonsFromShowingTheLowestPriceOfDiscountedProducts(iterable $taxons): void
+    {
+        $taxonsIris = [];
+        foreach ($taxons as $taxon) {
+            $taxonsIris[] = $this->iriConverter->getIriFromResource($taxon);
+        }
+
+        $this->client->addRequestData(
+            'channelPriceHistoryConfig',
+            ['taxonsExcludedFromShowingLowestPrice' => $taxonsIris],
+        );
+    }
+
+    /**
+     * @Then /^the "[^"]+" channel should have the lowest price of discounted products prior to the current discount (enabled|disabled)$/
+     */
+    public function theChannelShouldHaveTheLowestPriceOfDiscountedProductsPriorToTheCurrentDiscountEnabledOrDisabled(
+        string $visible,
+    ): void {
+        Assert::same(
+            $this->getChannelPricingFieldFromLastResponse('lowestPriceForDiscountedProductsVisible'),
+            $visible === 'enabled',
+        );
```

Comment:
```suggestion
        Assert::same(
            $this->getChannelPricingFieldFromLastResponse('lowestPriceForDiscountedProductsVisible'),
            $visible === 'enabled',
            sprintf('This channel should have the lowest price of discounted products prior to the current discount %s, but have not', $visible)
        );
```
lets add this error messages everywhere